src/Api/IngestionApiBase.php

```diff
@@ -49,12 +84,20 @@ public function getConfigSchema(): array
      */
     public function getRequestUriQuery(UriInterface $uri): array
     {
-        return [
+        $query = [
             'apiKey' => $this->getConfigValue('apiKey'),
             'database' => $this->getConfigValue('database'),
             'uri' => $this->getUri(),
-            'reference' => $this->getReference(),
         ] + parent::getRequestUriQuery($uri);
+
+        if ($languages = $this->getLanguages()) {
+            $query['language'] = $this->jsonEncoder->encode($languages, 'json');
+        }
+        if ($reference = $this->getReference()) {
+            $query['reference'] = $this->jsonEncoder->encode($reference, 'json');
```

Comment:
if the purpose is to encode we should use:
```suggestion
            $query['reference'] = rawurlencode($reference);
```
with json encode we will get double quoted strings like ""ref1""

Reply:
We need to pass the raw reference. It's already encoded in `ApiBase::getRequestUri()`.